libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2420,8 +2433,26 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	// TODO some members might be pure, but for example `address(0x123).balance` is not pure
 	// although every subexpression is, so leaving this limited for now.
 	if (auto tt = dynamic_cast<TypeType const*>(exprType))
+	{
 		if (tt->actualType()->category() == Type::Category::Enum)
 			annotation.isPure = true;
+		else if (tt->actualType()->category() == Type::Category::Contract)
+			if (auto const* ident = dynamic_cast<Identifier const*>(&_memberAccess.expression()))
+				if (ident->name() != "super")
+					if (auto const* funcType = dynamic_cast<FunctionType const*>(annotation.type))
+						if (funcType->hasDeclaration())
+							if (auto const* funcDef = dynamic_cast<FunctionDefinition const*>(&funcType->declaration()))
+								if (
+									!funcDef->isImplemented() &&
+									contains(m_scope->annotation().linearizedBaseContracts, funcDef->annotation().contract) &&
+									m_scopeFunction->name() == funcDef->name() &&
+									m_scopeFunction->parameters()  == funcDef->parameters()
+								)
+									m_errorReporter.typeError(
+										_memberAccess.location(),
+										"Referencing unimplemented function \"" + funcDef->name() + "\"."
```

Comment:
```suggestion
										"Referencing unimplemented function \"" + funcDef->name() + "\" in contract \"" + tt->actualType()->canonicalName() + "\"."
```

If you accept this change, test expectations need to be updated.